app/dashboard/static/js/app/view-tests-all.2020.5.js

```diff
@@ -43,62 +43,147 @@ require([
     var gSearchFilter;
     var gPageLen;
     var gTestsTable;
-    var gTestStatus = {};
+    var gTestCount = {};
+    var gDrawEventBound = false;
 
     setTimeout(function() {
         document.getElementById('li-test').setAttribute('class', 'active');
     }, 15);
 
     gDateRange = appconst.MAX_DATE_RANGE;
 
-    function getBatchStatusDone(results) {
+    function updateOrStageCount(opId, count) {
+        var element;
+
+        element = document.getElementById(opId);
+        // If we do not have the element in the DOM, it means dataTables has
+        // yet to add it.
+        if (element) {
+            html.replaceContent(
+                element, document.createTextNode(format.number(count)));
+
+            // Check if the data structure holding the data to update the
+            // elements still holds the element.
+            if (gTestCount.hasOwnProperty(opId)) {
+                delete gTestCount[opId];
+            }
+        } else {
+            // Store it in a dictionary for later access.
+            if (!gTestCount.hasOwnProperty(opId)) {
+                gTestCount[opId] = count;
+            }
+        }
```

Comment:
Are you sure this actually gets used?

When I loaded the page the first time, it took 12s to complete the batch query and there was no test counts shown until the query completed.  My understanding is that it should be showing the ones that can be drawn when it receives them, and store the other ones for later when the user goes to another page.

Reply:
Actually I think the logic here is not for showing results earlier, but to cope with the fact that some elements don't exist when they are on other pages.  So when the data arrives, only the rows that have been drawn can be updated.  The others get updated when the user goes to other pages on the table and other rows get drawn.

This is necessary but unrelated to the performance issue.